packages/clay-core/src/tree-view/useTree.ts

```diff
@@ -146,30 +162,39 @@ function createImmutableTree<T extends Array<Record<string, any>>>(tree: T) {
 				// location with the value that was removed.
 				case 'move': {
 					const nodeToRemove = nodeByPath(from);
-					const newChildren = nodeToRemove.parent.filter(
-						(item: any, index: number) =>
-							index !== nodeToRemove.index
-					);
-
-					let oldParentNode = tree[from[0]];
-					for (let i = 1; i < from.length - 2; i++) {
-						oldParentNode = oldParentNode.children[from[i]];
+
+					if (nodeToRemove.parent) {
+						nodeToRemove.parent[nestedKey] = nodeToRemove.parent[
+							nestedKey
+						].filter(
+							(item: any, index: number) =>
+								index !== nodeToRemove.index
+						);
+					} else {
+						immutableTree = immutableTree.filter(
```

Comment:
Hey @matuzalemsteles good job on the tests, however I'm seeing a "little" problem.
This is visible with the "dynamic" story, you'll see that when dragging a child node from the 3rd element (i.e. Documents and Media) to the 2nd item (i.e. Repositories) what happens is that the tree loses its first node (i.e. Liferay Drive). I added a test, but it passes, which either makes me think I'm testing the wrong thing, or using a totally different structure, so I'll review that part.
 
I also think we should always have a "parent", (by default it's the tree itself), because it's actually this branch of the `if/else` statement that is causing the bug. Anyway I think that with we're reaching our goal which means we're on the good way.

Reply:
@matuzalemsteles, the ["dynamic" story](https://github.com/liferay/clay/blob/2206cdb43b1de5b9fc4a15a04dd4cb718998a414/packages/clay-core/stories/TreeView.stories.tsx#L121-L156) is missing a `nestedKey` prop which is what was causing the issue. 

I've added it (locally for the moment), but maybe we should add a default value (`"children"` by default) if the user doesn't pass one, or always assume that `children` is going to be the nested key what do you think? 
This would be one step forward to make the API a bit stricter and "defined".

Reply:
@matuzalemsteles one last thing I've noticed is that with these changes, our ["page elements" story](https://github.com/liferay/clay/blob/2206cdb43b1de5b9fc4a15a04dd4cb718998a414/packages/clay-core/stories/TreeView.stories.tsx#L285-L403) is now "broken". I'm going to have a look at that now.

Reply:
@matuzalemsteles I think it might be because of [this](https://github.com/liferay/clay/blob/2206cdb43b1de5b9fc4a15a04dd4cb718998a414/packages/clay-core/src/tree-view/TreeViewItem.tsx#L36), and I also think that we could rename [`[left, right]`](https://github.com/liferay/clay/blob/2206cdb43b1de5b9fc4a15a04dd4cb718998a414/packages/clay-core/src/tree-view/TreeViewItem.tsx#L32) because it's not very clear. What do you think?

Reply:
> I've added it (locally for the moment), but maybe we should add a default value ("children" by default) if the user doesn't pass one, or always assume that children is going to be the nested key what do you think?

It makes sense to keep the default value for `children`.


> one last thing I've noticed is that with these changes, our "page elements" story is now "broken". I'm going to have a look at that now.

Looks like the dynamic examples broke, let me see what change broke that.

> I think it might be because of this, and I also think that we could rename [left, right] because it's not very clear. What do you think?

I'm not sure if that's what broke because this was already implemented well before making these changes to DnD, it was in the first PR of the sketch.

Well, the idea of `[left, right]` is that inside a TreeViewItem we can have two cases, the Stack (left) and the Group (right) but we also support the case of just having Stack, but we can change the name anyway, but nothing came. in my mind right now.

```jsx
<TreeView.Item>
  <TreeView.ItemStack>
    {...}
  </TreeView.ItemStack>
  <TreeView.Group>
    {...}
  </TreeView.Group>
</TreeView.Item>

// or just stack

<TreeView.Item>
  {...}
</TreeView.Item>
```

Reply:
I fix the bug, actually, it was my recent change, I had forgotten to pass the `nestedKey` to the context. https://github.com/liferay/clay/pull/4255/commits/8cfd75cd434c3e3f6e69196014b764044ac19d41